test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -358,32 +362,122 @@ defmodule Trento.SapSystemTest do
           tenant: tenant,
           instance_number: "00",
           instance_hostname: instance_hostname,
-          features: features,
+          features: "ABAP",
           http_port: http_port,
           https_port: https_port,
           start_priority: start_priority,
           host_id: host_id,
           health: :passing
         }),
         [
+          %ApplicationInstanceRegistered{
+            sap_system_id: sap_system_id,
+            sid: sid,
+            instance_number: "00",
+            instance_hostname: instance_hostname,
+            features: "ABAP",
+            http_port: http_port,
+            https_port: https_port,
+            start_priority: start_priority,
+            host_id: host_id,
+            health: :passing
+          },
           %SapSystemRegistered{
             sap_system_id: sap_system_id,
             sid: sid,
             db_host: db_host,
             tenant: tenant,
             health: :passing
-          },
+          }
+        ],
+        fn state ->
+          assert %SapSystem{
+                   sid: ^sid,
+                   application: %SapSystem.Application{
+                     sid: ^sid,
+                     instances: [
+                       %SapSystem.Instance{
+                         sid: ^sid,
+                         instance_number: "00",
+                         features: "ABAP",
+                         host_id: ^host_id,
+                         health: :passing
+                       },
+                       %SapSystem.Instance{
+                         features: "MESSAGESERVER"
+                       }
+                     ]
+                   }
+                 } = state
+        end
+      )
+    end
+
+    test "should register a SAP System and add an application instance when is already present a ABAP instance and one MESSAGESERVER instance is added" do
```

Comment:
`when an ABAP instance is already present and a new MESSAGESERVER instance is added`

Reply:
done